lib/hypothesis/possible.rb

```diff
@@ -0,0 +1,369 @@
+  # two names: A singular and a plural name. These are
+  # simply aliases and are identical in every way, but are
+  # provided to improve readability. For example
+  # `any integer` reads better than `given integers`
+  # but `arrays(of: integers)` reads better than
+  # `arrays(of: integer)`.
+  module Possibilities
+    include Hypothesis
+
+    class <<self
+      include Possibilities
+    end
+
+    # built_as lets you chain multiple Possible values together,
+    # by providing whatever value results from its block.
+    #
+    # For example the following provides a array plus some
+    # element from that array:
+    #
+    # ```ruby
+    #   built_as do
+    #     ls = any array(of: integers)
+    #     # Or min_size: 1 above, but this shows use of
+    #     # assume
+    #     assume ls.size > 0
+    #     i = any element_of(ls)
+    #     [ls, i]
+    # ```
+    #
+    # @return [Possible] A Possible whose possible values are
+    #   any result from the passed block.
+    def built_as(&block)
+      Hypothesis::Possible::Implementations::CompositePossible.new(block)
+    end
+
+    alias values_built_as built_as
+
+    # A Possible boolean value
+    # @return [Possible]
+    def booleans
+      integers(min: 0, max: 1).map { |i| i == 1 }
+    end
+
+    alias boolean booleans
+
+    # A Possible unicode codepoint.
+    # @return [Possible]
+    # @param min [Integer] The smallest codepoint to provide
+    # @param max [Integer] The largest codepoint to provide
+    def codepoints(min: 1, max: 1_114_111)
+      base = integers(min: min, max: max)
+      if min <= 126
+        from(integers(min: min, max: [126, max].min), base)
+      else
+        base
+      end
+    end
+
+    alias codepoint codepoints
+
+    # A Possible String
+    # @return [Possible]
+    # @param codepoints [Possible, nil] The Possible codepoints
+    #   that can be found in the string. If nil,
+    #   will default to self.codepoints. These
+    #   will be further filtered to ensure the generated string is
+    #   valid.
+    # @param min_size [Integer] The smallest valid length for a
+    #   provided string
+    # @param max_size [Integer] The smallest valid length for a
+    #   provided string
+    def strings(codepoints: nil, min_size: 0, max_size: 10)
+      codepoints = self.codepoints if codepoints.nil?
+      codepoints = codepoints.select do |i|
+        begin
+          [i].pack('U*').codepoints
+          true
+        rescue ArgumentError
+          false
+        end
+      end
+      arrays(of: codepoints, min_size: min_size, max_size: max_size).map do |ls|
+        ls.pack('U*')
+      end
+    end
+
+    alias string strings
+
+    # A Possible Hash, where all possible values have a fixed
+    # shape.
+    # This is used for hashes where you know exactly what the
+    # keys are, and different keys may have different possible values.
+    # For example, fixed_hashes(a: integers, b: booleans)
```

Comment:
Possibly a stale comment? Was `fixed_hashes` the old name for this function?

Ditto for `fixed_arrays` below.

Reply:
Yup, those were name changes where I failed to update the comment. Thanks, will fix.